R/method-differentialAbundance.R

```diff
@@ -53,34 +53,53 @@ setMethod("differentialAbundance", signature("AbsoluteAbundanceData"), function(
       veupathUtils::logWithTime("Replaced NAs with 0", verbose)
     }
 
+    # Err if the class of groupA does not match that of groupB
+    if (!identical(class(groupA), class(groupB))) {
+      stop("groupA and groupB must either both be character vectors or both be BinLists.")
+    }
+
+    # Determine if comparison variable is numeric.
+    isNumericComparisonVar <- identical(class(groupA)[1], "BinList")
+
     ## Check that groups are provided, if needed, and if they are provided,
     ## that they match at least one value in the comparisonVariable column.
     ## Eventually let's move this all into a Comparator class or similar.
     uniqueComparisonVariableValues <- sort(unique(sampleMetadata[[comparisonVariable]]))
-    
+
     if (!!length(groupA) && !!length(groupB)) {
-      # Does each group contain at least one string that matches a value in the comparisonValue column?
-      if (!any(groupA %in% uniqueComparisonVariableValues)) {
-        stop("At least one value in groupA must exist as a value in the comparisonValue sampleMetadata column.")
-      }
-      if (!any(groupB %in% uniqueComparisonVariableValues)) {
-        stop("At least one value in groupB must exist as a value in the comparisonValue sampleMetadata column.")
-      }
 
-      # Alert that we discard groupA/B values that are not found in the comparisonVariable
-      if (!all(groupA %in% uniqueComparisonVariableValues)) {
-        veupathUtils::logWithTime("Found values in groupA that do not exist in the comparisonVariable. Removing these values.", verbose)
-        groupA <- groupA[groupA %in% uniqueComparisonVariableValues]
-      }
-      if (!all(groupB %in% uniqueComparisonVariableValues)) {
-        veupathUtils::logWithTime("Found values in groupB that do not exist in the comparisonVariable. Removing these values.", verbose)
-        groupB <- groupB[groupB %in% uniqueComparisonVariableValues]
-      }
 
-      # Do not allow duplicated values
-      if (!!length(intersect(groupA, groupB))) {
-        veupathUtils::logWithTime("groupA and groupB cannot share members.", verbose)
-        stop()
+      if (!isNumericComparisonVar) {
+
+        # Do not allow duplicated values
```

Comment:
a lot of this feels like validation that could/ should belong to a `Comparator` class